### PR TITLE
Update tests to use default sc

### PR DESCRIPTION
**Pull request by jilju**
This changes have dependency on PR #965

### CODE REVIEW

**Comment by vasukulkarni** [comment]
@jilju if you can pickup the last commit and test, I think we should be good for merging both the PRs at the same time.

**Comment by jilju** [comment]
> @jilju if you can pickup the last commit and test, I think we should be good for merging both the PRs at the same time.

Last commit is picked and tests are updated accordingly.

**Comment by jilju** [comment]
> PR validation [Job UNSTABLE](https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/666//testReport) (some or all tests failed).

4 tests passed. 1 test failed
Failed test - test_pvc_delete_and_verify_size_is_returned_to_backend_pool

**Comment by jilju** [comment]
> PR validation [Job PASSED](https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/667//testReport).

Fixed test_pvc_delete_and_verify_size_is_returned_to_backend_pool.

**Comment by vasukulkarni** [comment]
I am restarting all tier1 to tier4 tests just to ensure once again nothing is broken, since this PR has my commits, it should help verify both.

**Comment by ebenahar** [comment]
@jilju Please rebase

**Comment by vasukulkarni** [comment]
@jilju lets merge this one along with #965, If you can rebase and run one final jenkins run please. Thanks!

**Comment by jilju** [comment]
> @jilju lets merge this one along with #965, If you can rebase and run one final jenkins run please. Thanks!

@vasukulkarni Please rebase #965 so that I can rebase this on top of it. There are conflicts with newly added functions. eg: storageclass_factory_session  from pr 1013. The function is being used in other places. I think that can be removed, but please confirm and rebase #965.

File: tests/conftest.py [hunk 1]
```diff
@@ -138,32 +142,6 @@ def finalizer():
     return factory
```
Comment (RazTamir) [inline tests/conftest.py]: I think we are doing something wrong here.
Removing the storage class fixtures is wrong IMO as we'll have use cases in the future needing this functionality

Comment (vasukulkarni) [reply tests/conftest.py]: Nothing is removed, functions still exists as we discussed before, so if some tests wants to use it they can still call those fixtures.

**Pull request merged**